### PR TITLE
[v6r20] Core: always close the socket even in case of exception

### PR DESCRIPTION
**Pull request by chaen**
Fixes https://github.com/DIRACGrid/DIRAC/issues/3905 
tested in LHCb prod environment
BEGINRELEASENOTES
*Core
FIX:  always close the socket even in case of exception

ENDRELEASENOTES

### CODE REVIEW

**Comment by andresailer** [comment]
Why not also change this in `sendErrorAndClose` a few lines above?

**Comment by chaen** [comment]
Because I did not see it :D
Good point, I add it

**Comment by chaen** [comment]
Done

**Comment by chaen** [comment]
I've added one more commit.
On heavily loaded servers, you can get system messages like 
```
kernel: TCPv6: Possible SYN flooding on port X.
```
There are various things to do to fix that, but among those are the need to change the socket backlog value in the `listen` call. 

This last commit allows to do it on a per service base. if you are happy with it, I will update the doc. Someone please answer :-)

File: Core/DISET/ServiceReactor.py [hunk 1]
```diff
@@ -34,123 +34,126 @@
...
-  __transportExtraKeywords = { 'SSLSessionTimeout' : False, 
-                               'IgnoreCRLs': False, 
-                               'PacketTimeout': 'timeout' }
+class ServiceReactor(object):
 
-  def __init__( self ):
+  __transportExtraKeywords = {'SSLSessionTimeout': False,
+                              'IgnoreCRLs': False,
+                              'PacketTimeout': 'timeout',
+                              'SocketBacklog': False,
```
Comment (fstagni) [inline Core/DISET/ServiceReactor.py]: Shouldn't this be a `int` value?

Comment (chaen) [reply Core/DISET/ServiceReactor.py]: No... This dict is just a lookup dictionary such that the "PacketTimeout" option is renamed "timeout" .....

**Comment by chaen** [comment]
I wanted to add somewhere the documentation about the new generic service option, but I do not know where. I do not think we have a place where we give the generic options. Do we ? @andresailer @atsareg @fstagni

**Pull request merged**